x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CopySettingsStep.java

```diff
@@ -10,34 +10,41 @@
 import org.apache.logging.log4j.Logger;
 import org.elasticsearch.cluster.ClusterState;
 import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.cluster.metadata.LifecycleExecutionState;
 import org.elasticsearch.cluster.metadata.Metadata;
 import org.elasticsearch.common.settings.Settings;
 import org.elasticsearch.index.Index;
 
 import java.util.Locale;
 import java.util.Objects;
+import java.util.function.BiFunction;
 
 /**
  * Copy the provided settings from the source to the target index.
  * <p>
- * The target index is derived from the source index using the provided prefix.
- * This is useful for actions like shrink or searchable snapshot that create a new index and migrate the ILM execution from the source
- * to the target index.
+ * The target index is generated by a supplier function.
+ * This is useful for actions like shrink, rollup or searchable snapshot that create
+ * a new index and migrate the ILM execution from the source to the target index.
  */
 public class CopySettingsStep extends ClusterStateActionStep {
     public static final String NAME = "copy-settings";
 
     private static final Logger logger = LogManager.getLogger(CopySettingsStep.class);
 
     private final String[] settingsKeys;
-    private final String indexPrefix;
 
-    public CopySettingsStep(StepKey key, StepKey nextStepKey, String indexPrefix, String... settingsKeys) {
+    private final BiFunction<String, LifecycleExecutionState, String> targetIndexNameSupplier;
```

Comment:
Now that I have removed the copy-settings step from rollup ILM action, I can revert the `CopySettingsStep` so that it `indexPrefix`. 

I only thought that making it more generic using an index name supplier function may be needed in the future. I don't have a strong opinion about keeping it this way or reverting it.

Reply:
I think it's fine to keep for now